### PR TITLE
SONARJAVA-3917 Add quick fixes for S1161 (OverrideAnnotationCheck)

### PR DESCRIPTION
**Pull request by m-g-sonar**

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/OverrideAnnotationCheck.java [hunk 1]
```diff
@@ -66,8 +81,62 @@ private static boolean isObjectMethod(Symbol.MethodSymbol method) {
...
+    }
+
+    Matcher paddingMatcher = PADDING_PATTERN.matcher(startOfLine);
+    if (paddingMatcher.find()) {
+      // use same spacing as the current line, potentially including mixed tabs and spaces
+      return newLineSequence + paddingMatcher.group(1);
+    }
+
+    // no whitespaces?
+    return newLineSequence;
```
Comment (quentin-jaquier-sonarsource) [inline java-checks/src/main/java/org/sonar/java/checks/OverrideAnnotationCheck.java:121]: In practice
```
  static class InnerClassSingleLine extends OverrideAnnotationCheckSimple { public void get() { } }
```
becomes (in my IDE)
```
  static class InnerClassSingleLine extends OverrideAnnotationCheckSimple {
    @Override
  public void get() { } }
```



If the method contains something before at the same line, we are already talking about strange formatted code, I don't think we should try to do something smart to cleanly add the code, a simple space (even without a new line) seems enough.

File: java-checks/src/main/java/org/sonar/java/checks/OverrideAnnotationCheck.java [hunk 2]
```diff
@@ -66,8 +81,62 @@ private static boolean isObjectMethod(Symbol.MethodSymbol method) {
...
   }
 
+  private String findNewLineSequence() {
+    String firstLine = QuickFixHelper.internalContext(context).getFileLinesWithLineEndings().get(0);
+    StringBuilder sb = new StringBuilder();
+    for (int i = firstLine.length() - 1; i == 0; i--) {
```
Comment (quentin-jaquier-sonarsource) [inline java-checks/src/main/java/org/sonar/java/checks/OverrideAnnotationCheck.java:127]: You have a problem with the condition of this loop!

**Review by quentin-jaquier-sonarsource — CHANGES_REQUESTED** [review]

File: java-checks-test-sources/src/main/java/checks/OverrideAnnotationCheck_QuickFixes.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+  interface I {
+    public void finalize() throws Throwable; // Compliant - from Object
+    void bar(); // Compliant
+  }
+
+  interface J extends I {
+    void bar(); // Compliant - from interface
+  }
+
```
[the remark this answers is not in the document]
Comment (m-g-sonar) [reply java-checks-test-sources/src/main/java/checks/OverrideAnnotationCheck_QuickFixes.java:18]: I actually need these parts to create the condition for the quick-fixes

[the remark this answers is not in the document]
Comment (m-g-sonar) [reply java-checks-test-sources/src/main/java/checks/OverrideAnnotationCheck_QuickFixes.java:18]: but indeed i can simplify a bit

**Review by quentin-jaquier-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**